ui/src/kapacitor/components/config/KafkaConfig.tsx

```diff
@@ -223,14 +283,37 @@ class KafkaConfig extends PureComponent<Props, State> {
       'ssl-cert': this.sslCert.value,
       'ssl-key': this.sslKey.value,
       'insecure-skip-verify': this.insecureSkipVerify.checked,
+      enabled: this.state.enabled,
+    }
+
+    if (this.isNewConfig) {
+      properties.id = this.id.value
     }
 
-    const success = await this.props.onSave(properties)
+    const success = await this.props.onSave(
+      properties,
+      this.isNewConfig,
+      this.configID
+    )
     if (success) {
       this.setState({testEnabled: true})
     }
   }
 
+  private handleEnabledChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
could you make this a bit more descriptive of a method name? i think a more conventional and descriptive name within this codebase for this would be `handleToggleEnabledCheckbox`, for example